### ISSUE TITLE
Symlinks to devices are not followed

### ISSUE THREAD

**Issue by Self-Perfection** [ddccontrol/ddccontrol#40]
My monitor does not have fixed address, it randomly chooses `/dev/i2c-5` or `/dev/i2c-4` on each boot. So I've written a custom udev rule to make permanent link:
```
KERNEL=="i2c-[0-9]*", SUBSYSTEM=="i2c-dev", SUBSYSTEMS=="i2c", ATTRS{name}=="i915 gmbus dpb", SYMLINK+="monitor-ddc"
```
It used to work well, but after upgrade to ddccontrol 20180508.r77978c7-1 it does not work anymore:
```
...
Reading EDID and initializing DDC/CI at bus dev:/dev/monitor-ddc...
Open monitor failed: GDBus.Error:ddccontrol.DDCControl.Error.InvalidDevice: only detected devices are allowed
.
DDC/CI at dev:/dev/monitor-ddc is unusable (-1).
...
```
Of course I can use a workaround like `ddccontrol "dev:$(readlink -f /dev/monitor)"`, but this still seems to be a bug for me.

**Comment by kravemir** [comment]
You're currently using unreleased version. If you want latest version, which should be stable, then use: https://github.com/ddccontrol/ddccontrol/releases/tag/0.4.4 . 

Thanks for a bug report for new version. The new version implements a D-Bus daemon to access monitors devices by a non-root user. As a security precaution, D-Bus daemon only allows detected monitors to be accessed (not any filename), see: https://github.com/ddccontrol/ddccontrol/blob/294d5c583631cc331d06d296f7821ff8cee9294c/src/daemon/service.c#L148

And : https://github.com/ddccontrol/ddccontrol/blob/7a7cdb454697e920fb11479b531d36f31b70b609/src/lib/ddcci.c#L1217-L1245

I was thinking to add support for daemon configuration, ie. `/etc/ddccontrol-service.conf`, with contest like:

```
probe_dirs=/dev
probe_whitelist=/dev/i2c.* /dev/monitor-ddc
```
Do you have any ideas, on this? Or, do you think following of symlinks, before check, whether a device can be opened, would be fine?

**Comment by Self-Perfection** [comment]
Yes, I've noticed new D-Bus daemon. And it is cool, with daemon setting brightness is 4 times faster and does not annoy me anymore!

In my case following symlinks certainly would suffice and it seems to be generally expected behaviour. Config file for daemon seems unnecessary complication for me. How one might end up with device node not adhering to naming conventions or outside of `/dev/` directory? Or why would anybody want to blacklist device from probing?

**Comment by kravemir** [comment]
> In my case following symlinks certainly would suffice and it seems to be generally expected behaviour. Config file for daemon seems unnecessary complication for me. How one might end up with device node not adhering to naming conventions or outside of /dev/ directory? Or why would anybody want to blacklist device from probing?

It's not blacklist, but whitelist. I'm just taking pessimistic approach, because daemon runs under root, and I don't want a non-root user to be able to touch any root-owned file with the daemon.
